generators/Exercise.cs

```diff
@@ -11,7 +11,7 @@ public abstract class Exercise
         
         public string Name => GetType().ToExerciseName();
 
-        protected CanonicalData CanonicalData { get; private set; }
+        private CanonicalData CanonicalData { get; set; }
```

Comment:
Could you make it `private CanonicalData _canonicalData` and move it to the top of the class to conform to the rest of the code?